frontend/packages/dev-console/src/components/monitoring/metrics/MonitoringMetrics.tsx

```diff
@@ -1,9 +1,25 @@
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { getURLSearchParams } from '@console/internal/components/utils';
+import { queryBrowserRunQueries, queryBrowserPatchQuery } from '@console/internal/actions/ui';
+import { QueryObj } from '@console/internal/components/monitoring/query-browser';
 import MetricsDropdown from './MetricsDropdown';
 import MetricsChart from './MetricsChart';
 
-const MonitoringMetrics: React.FC = () => {
+type MonitoringMetricsProps = {
+  patchQuery: (patch: QueryObj) => void;
+  runQueries: () => never;
+};
+
+const MonitoringMetrics: React.FC<MonitoringMetricsProps> = ({ patchQuery, runQueries }) => {
+  const params = getURLSearchParams();
+
+  React.useEffect(() => {
+    patchQuery({ text: params.query0 });
+    runQueries();
+  }, [params, patchQuery, runQueries]);
```

Comment:
can we extract it above and just have `query0` as dependency to `useEffect` instead of `params`